CMakeLists.txt

```diff
@@ -77,95 +61,145 @@ elseif(${CMAKE_BUILD_TARGET} MATCHES "Both")
+  )
+  if(ENABLE_UVC_DEBUGGING)
+    target_compile_definitions(${target_name}
+      PRIVATE
+        UVC_DEBUGGING
+    )
+  endif()
 endforeach()
 
-if(BUILD_UVC_SHARED)
-  if(JPEG_FOUND)
-    target_link_libraries (uvc ${JPEG_LINK_FLAGS})
-  endif(JPEG_FOUND)
-
-  target_link_libraries(uvc ${LIBUSB_LIBRARIES})
-
-  #add_executable(test src/test.c)
-  #target_link_libraries(test uvc ${LIBUSB_LIBRARIES} opencv_highgui
-  #  opencv_core)
-endif()
-
 if(BUILD_EXAMPLE)
   add_executable(example src/example.c)
-  target_link_libraries(example uvc pthread)
+  find_package(Threads)
+  target_link_libraries(example
+    PRIVATE
+      LibUVC::UVC
+      Threads::Threads
+  )
 endif()
 
 if(BUILD_TEST)
-
-    # Try to find OpenCV using a module or pkg-config. If that doesn't work, search for the header.
-  find_package(opencv QUIET)
-  if(OPENCV_FOUND)
-    set(OPENCV_LINK_FLAGS ${OPENCV_LIBRARIES})
-  else()
-    pkg_check_modules(OPENCV QUIET opencv)
-    if(OPENCV_FOUND)
-        set(OPENCV_INCLUDE_DIR ${OPENCV_INCLUDE_DIRS})
-        set(OPENCV_LINK_FLAGS ${OPENCV_LDFLAGS})
-    endif()
-  endif()
+  find_package(OpenCV
+    REQUIRED COMPONENTS
+      opencv_core
+      opencv_highgui
+  )
 
   add_executable(uvc_test src/test.c)
-  target_link_libraries(uvc_test uvc ${OPENCV_LDFLAGS})
+  target_link_libraries(uvc_test
+    PRIVATE
+      LibUVC::UVC
+      opencv_core
+      opencv_highgui
+  )
 endif()
 
-install(TARGETS ${UVC_TARGETS}
-    EXPORT libuvcTargets
-  LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}"
-  ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}"
-  PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_PREFIX}/include/libuvc"
-)
-
-export(TARGETS ${UVC_TARGETS}
-  FILE "${PROJECT_BINARY_DIR}/libuvcTargets.cmake")
-export(PACKAGE libuvc)
 
-set(CONF_INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/include")
-set(CONF_LIBRARY_DIR "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}")
-set(CONF_LIBRARY "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}/${CMAKE_SHARED_LIBRARY_PREFIX}uvc${CMAKE_SHARED_LIBRARY_SUFFIX}")
+include(GNUInstallDirs)
+set(CMAKE_INSTALL_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/libuvc)
+
+install(
+  TARGETS ${UVC_TARGETS}
+  EXPORT libuvcTargets
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libuvc
+)
 
-configure_file(libuvcConfig.cmake.in ${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/libuvcConfig.cmake)
+install(EXPORT libuvcTargets
+  FILE libuvcTargets.cmake
+  NAMESPACE LibUVC::
+  DESTINATION ${CMAKE_INSTALL_CMAKEDIR}
+)
 
-configure_file(libuvcConfigVersion.cmake.in ${PROJECT_BINARY_DIR}/libuvcConfigVersion.cmake @ONLY)
+install(FILES 
+    cmake/FindLibUSB.cmake
+    cmake/FindJpegPkg.cmake
+  DESTINATION ${CMAKE_INSTALL_CMAKEDIR}
```

Comment:
Since when we use libuvc in our code we still need to search its dependencies JPEG and libusb-1.0. So we copy scripts for `find_package` command inside `libuvcConfig.cmake` file.